### PR TITLE
logic for guardrails around haproxy reloads

### PR DESCRIPTION
**Pull request by sawall**

### CODE REVIEW

File: ansible/roles/hcv-haproxy-configure/files/haproxy-configurator-payload.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# payload of the haproxy-configurator, assumed to be an operation protected by a consul lock
+
+echo -n "jitsi.haproxy.reconfig_locked:1|c" | nc -4u -w1 localhost 8125
```
Comment (aaronkvanmeerten) [inline ansible/roles/hcv-haproxy-configure/files/haproxy-configurator-payload.sh:5]: Maybe this belongs outside the payload but inside the outer lock logic?

File: ansible/roles/hcv-haproxy-configure/files/haproxy-configurator-payload.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    echo "#### hcp: haproxy failed to undrain from the load balancer" | tee -a $TEMPLATE_LOGFILE
+    FINAL_EXIT=1
+fi
+
+# emit success metric and log
+echo -n "jitsi.haproxy.reconfig:1|c" | nc -4u -w1 localhost 8125
+echo "#### hcp: succeeded to reload haproxy with new config" >> $TEMPLATE_LOGFILE
+fi
+# obtain a consul lock and install the haproxy config
+FINAL_EXIT=consul lock haproxy_reload "/usr/local/bin/haproxy-cfg-install.sh"
```
Comment (aaronkvanmeerten) [inline ansible/roles/hcv-haproxy-configure/files/haproxy-configurator-payload.sh:37]: Isn't this what the cp and service reload above is doing?  and why is this where the consul lock is taken?  inside the payload?

File: ansible/roles/hcv-haproxy-configure/files/haproxy-configurator.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    echo "#### hc: $DRAFT_CONFIG; is identical to /etc/haproxy/haproxy.cfg, exiting configurator " >> $TEMPLATE_LOGFILE
+    exit 0
+fi
+
+while true; do
+    UNIX_TIME=$(date +%s)
+    UNIX_TIME_OF_VALIDATED_CONFIG_FILE=$(stat -c %Y "${DRAFT_CONFIG}.validated")
+    if [ $UNIX_TIME_OF_VALIDATED_CONFIG_FILE + 60 -gt $UNIX_TIME ]; then
+        echo "#### hc: validated config file has been stable for 60 seconds, initiating locked reload" >> $TEMPLATE_LOGFILE
+        consul lock -timeout=10m haproxy-configurator-payload.sh
```
Comment (aaronkvanmeerten) [inline ansible/roles/hcv-haproxy-configure/files/haproxy-configurator.sh:16]: This seems like the right place for the lock, indeed.

**Review by aaronkvanmeerten — COMMENTED** [review]
Not sure about some of the above, so just commenting for now.

File: ansible/roles/hcv-haproxy-configure/files/haproxy-configurator-payload.sh (hunk 2 above)
Comment (sawall) [reply ansible/roles/hcv-haproxy-configure/files/haproxy-configurator-payload.sh:37]: oops, fixed

**Review by aaronkvanmeerten — APPROVED** [review]
looks good enough to test out

**Review by aaronkvanmeerten — APPROVED** [review]

**Pull request merged**